pkg/apis/core/validation/validation.go

```diff
@@ -2767,6 +2767,10 @@ func validateTCPSocketAction(tcp *core.TCPSocketAction, fldPath *field.Path) fie
 	return ValidatePortNumOrName(tcp.Port, fldPath.Child("port"))
 }
 
+func validateGRPCAction(grpc *core.GRPCAction, fldPath *field.Path) field.ErrorList {
+	return ValidatePortNumOrName(intstr.FromInt(int(grpc.Port)), fldPath.Child("port"))
+}
+
 func validateHandler(handler *core.Handler, fldPath *field.Path) field.ErrorList {
```

Comment:
can you pls add a parameter that validates that this is called from `validateProbe` and only than allow gRPC

Reply:
A little confused on this review comment, can you make it more clear?

Reply:
since gRPC only supported as a probe, I'd suggest validate to fail if grpc is used as lifecycle handler